policies/metrics_collisions.rego

```diff
@@ -0,0 +1,43 @@
+package after_resolution
+
+import rego.v1
+
+# Data structures to make checking things faster.
+metric_names := { obj |
+    group := input.groups[_];
+    group.type = "metric"
+    obj := { "name": group.metric_name, "namespace_prefix": extract_metric_namespace_prefix(group.metric_name)}
+}
+
+# check that matric names do not collide with namespaces
+deny contains metrics_registry_collision(description, name) if {
+    some i
+
+    name := metric_names[i].name
+    prefix := metric_names[i].namespace_prefix
```

Comment:
let's exclude deprecated metrics - it should cover most, if not all, of the existing collisions.

Same on entities - it's ok to have collision with a deprecated thing

Reply:
true, missed that attribute, thanks!